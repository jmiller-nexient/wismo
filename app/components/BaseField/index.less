@import '../../styles/themes/index.less';

@base-field-prefix-cls: ~'@{dte-prefix}-base-field';

.@{base-field-prefix-cls} {
  &-section {
    display: inline-block;
    margin: 15px;
    width: 50%;

    @media screen and (max-width: 400px) {
      width: 100%;
    }
  }

  &-label {
    font-size: @dte-wismo-font-size-2;
    font-weight: @dte-wismo-font-weight-semi-bold;
  }

  &-children {
    color: @dte-gray-500;
  }
}
