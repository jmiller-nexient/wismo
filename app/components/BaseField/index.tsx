import React, { ReactNode } from 'react';
import { FormattedMessage, MessageDescriptor } from 'react-intl';

import './index.less';

interface IBaseFieldProps {
  children: ReactNode;
  label: MessageDescriptor;
}

const BaseField: React.FC<IBaseFieldProps> = (props: IBaseFieldProps): React.ReactElement => {
  const { label } = props;

  return (
    <div className={'dte-wismo-base-field-section'}>
      <label className={'dte-wismo-base-field-label'}>
        <FormattedMessage {...label} />
      </label>
      <div>
        <span className={'dte-wismo-base-field-children'}>
          {React.Children.toArray(props.children)}
        </span>
      </div>
    </div>
  );
};

export default BaseField;
