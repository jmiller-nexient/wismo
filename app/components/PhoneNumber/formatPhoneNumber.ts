export default function formatPhoneNumber(value: string) {

    const formattedPhoneNumber = value.length === 7 ? `${value.slice(0, 3)}-${value.slice(3, value.length)}`
        :
        `(${value.slice(0, 3)}) ${value.slice(3, 6)}-${value.slice(6, value.length)}`;

    return formattedPhoneNumber;
}
