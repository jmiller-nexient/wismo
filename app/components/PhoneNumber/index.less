@import '../../styles/themes/index.less';

.dte-wismo-phone-number {
  color: @dte-gray-500;
  font-family: @dte-wismo-font-family;
  font-size: @dte-wismo-font-size-1;
}
