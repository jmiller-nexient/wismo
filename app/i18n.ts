/**
 * i18n.js
 *
 * This will setup the i18n language files and locale data for your app.
 *
 *   IMPORTANT: This file is used by the internal build
 *   script `extract-intl`, and must use CommonJS module syntax
 *   You CANNOT use import/export in this file.
 */
import { DEFAULT_LOCALE } from './locales';

const enTranslationMessages = require('translations/en.json');
const esTranslationMessages = require('translations/es.json');

// The purpose of this method is to apply fallback logic when a translation
// is not defined for the current locale.
// In our case, we do not want any fallback logic.
// tslint:disable-next-line: typedef
const formatTranslationMessages = (locale: string, messages) => {
  const defaultFormattedMessages =
    locale !== DEFAULT_LOCALE ? formatTranslationMessages(DEFAULT_LOCALE, enTranslationMessages) : {};

  // tslint:disable-next-line: typedef
  const flattenFormattedMessages = (formattedMessages, key) => {
    const formattedMessage =
      !messages[key] && locale !== DEFAULT_LOCALE ? defaultFormattedMessages[key] : messages[key];

    return { ...formattedMessages, [key]: formattedMessage };
  };

  return Object.keys(messages).reduce(flattenFormattedMessages, {});
};

export const translationMessages = {
  en: formatTranslationMessages('en', enTranslationMessages),
  es: formatTranslationMessages('es', esTranslationMessages),
};
