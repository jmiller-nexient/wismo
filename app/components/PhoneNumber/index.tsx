import * as React from 'react';

import formatPhoneNumber from './formatPhoneNumber';
import './index.less';

interface IPhoneNumberProps {
  value: string;
}

const PhoneNumber: React.FC<IPhoneNumberProps> = (props: IPhoneNumberProps) => {
  const { value } = props;

  if (value.length !== 10 && value.length !== 7) {
    throw new Error('phone numbers must be 10 or 7 numeric digits.');
  }

  const formattedPhoneNumber = formatPhoneNumber(value);

  return (
    <span className="dte-wismo-phone-number">
      {formattedPhoneNumber}
    </span>
  );
};

export default PhoneNumber;
