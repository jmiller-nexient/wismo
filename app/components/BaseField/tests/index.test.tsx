import React from 'react';
import { IntlProvider, MessageDescriptor } from 'react-intl';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import { translationMessages } from '../../../i18n';

import BaseField from 'components/BaseField';
import PhoneNumber from 'components/PhoneNumber';

const label: MessageDescriptor = {
  id: 'app.base-form-field-label',
  defaultMessage: 'This is just a filler message',
};

const fillerComponent = <PhoneNumber value={'1234567890'}/>;

describe('<BaseField />', () => {
  it('should render correctly', () => {
    const tree: ReactTestRendererJSON | null = renderer.create(
      <IntlProvider locale="en" messages={translationMessages.en}>
        <BaseField children={fillerComponent} label={label}/>
      </IntlProvider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
