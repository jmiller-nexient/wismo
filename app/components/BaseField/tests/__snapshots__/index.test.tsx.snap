// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseField /> should render correctly 1`] = `
<div
  className="dte-wismo-base-field-section"
>
  <label
    className="dte-wismo-base-field-label"
  >
    This is just a filler message
  </label>
  <div>
    <span
      className="dte-wismo-base-field-children"
    >
      <span>
        (123) 456-7890
      </span>
    </span>
  </div>
</div>
`;
